src/controllers/auth.js

```diff
@@ -0,0 +1,53 @@
+const User = require("../models/User");
+const { StatusCodes } = require("http-status-codes");
+const { BadRequestError, UnauthenticatedError } = require("../errors");
+
+//const jwt = require("jsonwebtoken");
+
+const register = async (req, res) => {
+  // can do validation vis mangoos in the controller dirrectly
+  // const { name, email, password } = req.body;
+  // if (!name || !email || !password) {
+  //   throw new BadRequestError("Please provide name, password, and email");
+  // }
+
+  const user = await User.create({ ...req.body });
+  //const token = jwt.sign({ userId: user._id, name: user.name }, "jwtSecret", {
+  //   expiresIn: "30d",
+  // }); //user.createJWT()
+
+  console.log("test");
+  console.log("name ", user.name);
+  const token = user.createJWT();
```

Comment:
add createJWT function to User model file

Reply:
added